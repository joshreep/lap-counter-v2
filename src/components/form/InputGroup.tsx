import { FC, InputHTMLAttributes, Ref } from 'react'
import { DisabledInput, DisabledInputText, Group, Input, Label } from './InputGroup.style'

interface InputGroupProps extends InputHTMLAttributes<HTMLInputElement> {
  id: string
  label: string
  ref?: Ref<HTMLInputElement>
  'data-testid'?: string
}

const InputGroup: FC<InputGroupProps> = ({ disabled, label, ref, ...props }) => {
  return (
    <Group>
      <Label htmlFor={props.id} disabled={disabled}>
        {label}
      </Label>
      {disabled ? (
        <DisabledInput>
          <DisabledInputText data-testid={props['data-testid']}>{props.value}</DisabledInputText>
        </DisabledInput>
      ) : (
        <Input {...props} ref={ref} />
      )}
    </Group>
  )
}

export default InputGroup
