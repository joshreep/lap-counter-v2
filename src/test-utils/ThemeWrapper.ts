import MatchMediaMock from 'jest-matchmedia-mock'

type Theme = 'light' | 'dark'

const themes: Theme[] = ['light', 'dark']

export function forEachTheme(callback: (theme: Theme) => void) {
  themes.forEach((theme) => {
    const matchMedia = new MatchMediaMock()
    matchMedia.useMediaQuery(`(prefers-color-scheme: ${theme})`)

    callback(theme)

    matchMedia.clear()
  })
}
