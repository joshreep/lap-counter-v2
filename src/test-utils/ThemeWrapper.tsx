import SCDarkTheme from '@/theme/dark-theme'
import SCLightTheme from '@/theme/light-theme'
import {
  DarkTheme as RNDarkTheme,
  DefaultTheme as RNDefaultTheme,
  ThemeProvider as RNThemeProvider,
} from '@react-navigation/native'
import { FC, PropsWithChildren } from 'react'
import { ColorSchemeName } from 'react-native'
import { ThemeProvider as SCThemeProvider } from 'styled-components/native'

export const themes: ColorSchemeName[] = ['light', 'dark']

export function getThemeWrapper(theme: ColorSchemeName) {
  const Wrapper: FC<PropsWithChildren> = ({ children }) => {
    return (
      <RNThemeProvider value={theme === 'dark' ? RNDarkTheme : RNDefaultTheme}>
        <SCThemeProvider theme={theme === 'dark' ? SCDarkTheme : SCLightTheme}>{children}</SCThemeProvider>
      </RNThemeProvider>
    )
  }

  return Wrapper
}

type ForEachThemeCallback = (theme: ColorSchemeName, wrapper: FC<PropsWithChildren>) => void

export function forEachTheme(callback: ForEachThemeCallback) {
  themes.forEach((theme) => {
    callback(theme, getThemeWrapper(theme))
  })
}
