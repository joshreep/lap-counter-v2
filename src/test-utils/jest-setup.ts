import '@testing-library/react-native/extend-expect'
import '@testing-library/jest-native/extend-expect'

jest.mock('react-native-reanimated', () => null, { virtual: true })
jest.mock('@testing-library/jest-native/extend-expect', () => null, { virtual: true })
