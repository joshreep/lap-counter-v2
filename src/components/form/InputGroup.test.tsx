import { getThemeWrapper, themes } from '@/test-utils/ThemeWrapper'
import InputGroup from './InputGroup'
import { render } from '@testing-library/react-native'

themes.forEach((theme) => {
  const wrapper = getThemeWrapper(theme)

  describe(`${theme} theme`, () => {
    test('should match snapshot', () => {
      const container = render(<InputGroup label="name" value="George Washington" />, { wrapper })
      expect(container.toJSON()).toMatchSnapshot()
    })

    test('should match snapshot when disabled', () => {
      const container = render(<InputGroup label="birthday" value="7/4/1776" disabled />, { wrapper })
      expect(container.toJSON()).toMatchSnapshot()
    })
  })
})
