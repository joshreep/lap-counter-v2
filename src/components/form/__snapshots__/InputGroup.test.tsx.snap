// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`dark theme should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "gap": 4,
    }
  }
>
  <Text
    style={
      {
        "color": "rgb(255, 255, 255)",
        "fontSize": 16,
        "marginLeft": 5,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
  >
    name
  </Text>
  <TextInput
    clearButtonMode="while-editing"
    style={
      {
        "backgroundColor": "rgba(255,255,255, .1)",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "color": "rgb(255, 255, 255)",
        "fontSize": 16,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
        "width": "100%",
      }
    }
    value="George Washington"
  />
</View>
`;

exports[`dark theme should match snapshot when disabled 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "gap": 4,
    }
  }
>
  <Text
    disabled={true}
    style={
      {
        "color": "rgb(127, 127, 127)",
        "fontSize": 16,
        "marginLeft": 5,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
  >
    birthday
  </Text>
  <View
    style={
      {
        "backgroundColor": "rgba(255, 255, 255, .05)",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "rgb(127, 127, 127)",
          "fontSize": 16,
        }
      }
    >
      7/4/1776
    </Text>
  </View>
</View>
`;

exports[`light theme should match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "gap": 4,
    }
  }
>
  <Text
    style={
      {
        "color": "#111",
        "fontSize": 16,
        "marginLeft": 5,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
  >
    name
  </Text>
  <TextInput
    clearButtonMode="while-editing"
    style={
      {
        "backgroundColor": "rgba(0, 0, 0, .07)",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "color": "#111",
        "fontSize": 16,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
        "width": "100%",
      }
    }
    value="George Washington"
  />
</View>
`;

exports[`light theme should match snapshot when disabled 1`] = `
<View
  style={
    {
      "backgroundColor": "transparent",
      "gap": 4,
    }
  }
>
  <Text
    disabled={true}
    style={
      {
        "color": "rgb(127, 127, 127)",
        "fontSize": 16,
        "marginLeft": 5,
        "paddingBottom": 0,
        "paddingLeft": 0,
        "paddingRight": 0,
        "paddingTop": 0,
      }
    }
  >
    birthday
  </Text>
  <View
    style={
      {
        "backgroundColor": "rgba(0, 0, 0, .035)",
        "borderBottomLeftRadius": 4,
        "borderBottomRightRadius": 4,
        "borderTopLeftRadius": 4,
        "borderTopRightRadius": 4,
        "paddingBottom": 15,
        "paddingLeft": 15,
        "paddingRight": 15,
        "paddingTop": 15,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        {
          "color": "rgb(127, 127, 127)",
          "fontSize": 16,
        }
      }
    >
      7/4/1776
    </Text>
  </View>
</View>
`;
